### PR TITLE
update Mathlib

### PR DESCRIPTION
**Pull request by LorenzoLuccioli**
update Mathlib to the latest version, no fixes needed

### CODE REVIEW

**Comment by RemyDegenne** [comment]
No fixes are needed, but we have a bunch of lemmas that duplicate those newly added to Mathlib, no? Could you remove all of those?

**Comment by RemyDegenne** [comment]
Also the build does not succeed, so there are apparently fixes needed :)
And the build log also highlights a bunch of deprecated lemma uses: we should switch to their replacements.

**Comment by LorenzoLuccioli** [comment]
> No fixes are needed, but we have a bunch of lemmas that duplicate those newly added to Mathlib, no? Could you remove all of those?

The new lemmas in Mathlib from my last 3 PRs are not yet in the upstream, I still have to PR those, so I thought that I should bump first and then PR them to upstream already fixed for the new version of Mathlib. Are there other changes that I'm missing?

**Comment by RemyDegenne** [comment]
The whole L1Space file was merged to Mathlib earlier, wasn't it?

**Comment by LorenzoLuccioli** [comment]
> Also the build does not succeed, so there are apparently fixes needed :) And the build log also highlights a bunch of deprecated lemma uses: we should switch to their replacements.

Ok, now I should have fixed it , the fail was caused by an 'apply? ... says ...', so I could not see that in local.

**Comment by LorenzoLuccioli** [comment]
> The whole L1Space file was merged to Mathlib earlier, wasn't it?

Yes, right. Now that should be updated.

File: TestingLowerBounds/KullbackLeibler.lean [hunk 1]
```diff
@@ -124,7 +123,8 @@ lemma kl_ge_mul_log (μ ν : Measure α) [IsFiniteMeasure μ] [IsFiniteMeasure 
   by_cases hν : ν = 0
   · refine absurd ?_ hμ
     rw [hν] at hμν
-    apply? says exact Measure.measure_univ_eq_zero.mp (hμν rfl)
+    set_option says.verify true in
```
Comment (RemyDegenne) [inline TestingLowerBounds/KullbackLeibler.lean]: I don't know what that line means. What does it do?

Regardless of the explanation: we could remove that line.

Comment (LorenzoLuccioli) [reply TestingLowerBounds/KullbackLeibler.lean]: Right, it was suggested by the fail log before in order to reproduce locally the error with the 'apply? ... says ...'.

**Comment by LorenzoLuccioli** [comment]
> And the build log also highlights a bunch of deprecated lemma uses: we should switch to their replacements.

With this do you mean this message:
'Node.js 16 actions are deprecated. Please update the following actions to use Node.js 20: actions/checkout@v2. For more information see: https://github.blog/changelog/2023-09-22-github-actions-transitioning-from-node-16-to-node-20/.'?

**Review by RemyDegenne — APPROVED** [review]

**Pull request merged**